crates/abd-clam/src/core/graph/_graph.rs

```diff
@@ -7,6 +7,7 @@ use distances::Number;
 // use crate::core::{cluster::Cluster, dataset::VecDataset};
 use crate::core::graph::criteria::{detect_edges, select_clusters};
 
+use crate::VecDataset;
```

Comment:
Why is this import here? I would prefer all methods to be generic over all datasets